src/utils/imageNameUtils.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { ContainerRegistryManagementClient, ContainerRegistryManagementModels } from "@azure/arm-containerregistry";
+import { ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { acrDomain, acrDomainRegExp, dockerHubDomain, dockerHubDomainRegExp, SupportedRegistries } from "../constants";
+import { createContainerRegistryManagementClient } from "./azureClients";
+import { localize } from "./localize";
+
+export namespace imageNameUtils {
+    export function detectRegistryDomain(imageName: string): SupportedRegistries | undefined {
```

Comment:
I'm a little leery of the regex approach here, since there's potential for mistakes. For example, if I supply an image with the name "azurecr.io.foo/bar:baz", it would be incorrectly detected as belonging to ACR. True, this example is contrived, but my gut is telling me something else could happen.

We _can_ supply the registry domain specifically, as part of the contract. The Docker extension already has some fairly sophisticated image name parsers that we trust. Given the domain, a more exact regex could be applied, such as `/\.azurecr\.io$/i` (which demands "azurecr.io" be the end of the domain, and that it start with a dot, e.g. "myfoo.azurecr.io" passes, but "myfoo.azurecr.iot" does not). For Docker Hub, I think `/^docker\.io$/i` is sufficient.

Reply:
Let me know if you think something like this would be ok!

![image](https://user-images.githubusercontent.com/40250218/213816580-fe8549ff-6657-4ef8-a49c-228dc1bf6fea.png)

Instead of amending the contract, just parsing the `registryName` off of the `imageName` and doing the regex search on that value.

Reply:
Instead of `loginServer`, could we just send `registry` (the domain), and for Docker Hub's case just map `loginServer` to the other value it wants? IIRC, for ACR and others the login server === registry domain right?